### PR TITLE
Use label for prop in Select [required for mui>4.9.0]

### PR DESCRIPTION
**Pull request by afzalsayed96**
This passes the label prop to the Select component.

Refer mui-org/material-ui/issues/19144

### CODE REVIEW

**Comment by lookfirst** [comment]
So, first off... thank you so much for the PR!

Unfortunately, this is causing the test failures. Before you submit PR's for tested projects like this one, you should always run the tests first and get everything happy. =)

This also isn't the right fix. The issue you linked is very helpful. It sounds like they fixed the whole width issue. So, please read their updated documentation, make the appropriate changes and submit an updated PR that also passes tests.

Thank you!

**Comment by afzalsayed96** [comment]
Thanks for the comment @lookfirst 
I did read their updated documentation. It requires to remove labelWidth and provide only label prop. However I assumed it would be necessary for people using older versions.

Could you lend some help in fixing the test. After changing line 95 of `src/Select.tsx` to
```diff
<MuiSelect
	name={name}
	value={value}
	onChange={onChange}
	multiple={multiple}
-       label
+ 	label={!!label ? label : undefined}
```


the test looks like this



```diff
$ tsdx test
 PASS  test/TextField.test.tsx
 PASS  test/Checkboxes.test.tsx
 FAIL  test/Select.test.tsx

  ● Select › basic component › renders outlined

    expect(received).toMatchSnapshot()

    Snapshot name: `Select basic component renders outlined 1`

    - Snapshot
    + Received

    @@ -52,18 +52,16 @@
                    />
                  </svg>
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-81 MuiOutlinedInput-notchedOutline-44"
-                   style="padding-left: 8px;"
                  >
                    <legend
-                     class="PrivateNotchedOutline-legend-82"
-                     style="width: 0.01px;"
+                     class="PrivateNotchedOutline-legendLabelled-83"
                    >
                      <span>
-    
+                       Test
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
    @@ -121,18 +119,16 @@
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-81 MuiOutlinedInput-notchedOutline-44"
-                 style="padding-left: 8px;"
                >
                  <legend
-                   class="PrivateNotchedOutline-legend-82"
-                   style="width: 0.01px;"
+                   class="PrivateNotchedOutline-legendLabelled-83"
                  >
                    <span>
-    
+                     Test
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>

      108 | 					<SelectComponent data={selectData} initialValues={initialValues} variant="outlined" label={true} />
      109 | 				);
    > 110 | 				expect(rendered).toMatchSnapshot();
          | 				                 ^
      111 | 				console.log('outlined', true);
      112 | 			});
      113 | 		});

      at test/Select.test.tsx:110:22
      at step (node_modules/tslib/tslib.js:139:27)
      at Object.next (node_modules/tslib/tslib.js:120:57)
      at node_modules/tslib/tslib.js:113:75
      at Object.__awaiter (node_modules/tslib/tslib.js:109:16)
      at test/Select.test.tsx:106:14
      at batchedUpdates$1 (node_modules/react-dom/cjs/react-dom.development.js:21887:12)
      at Object.act (node_modules/react-dom/cjs/react-dom-test-utils.development.js:929:14)
      at test/Select.test.tsx:106:10
      at step (node_modules/tslib/tslib.js:139:27)
      at Object.next (node_modules/tslib/tslib.js:120:57)
      at node_modules/tslib/tslib.js:113:75
      at Object.__awaiter (node_modules/tslib/tslib.js:109:16)
      at Object.<anonymous> (test/Select.test.tsx:105:26)

 › 1 snapshot failed.
 PASS  test/Radios.test.tsx
 PASS  test/KeyboardDatePicker.test.tsx
 PASS  test/Switches.test.tsx
 PASS  test/TimePicker.test.tsx
 PASS  test/DatePicker.test.tsx
 PASS  test/KeyboardTimePicker.test.tsx
 PASS  test/Autocomplete.test.tsx

Snapshot Summary
 › 1 snapshot failed from 1 test suite. Inspect your code changes or re-run jest with `-u` to update them.

Test Suites: 1 failed, 9 passed, 10 total
Tests:       1 failed, 91 passed, 92 total
Snapshots:   1 failed, 62 passed, 63 total
Time:        6.333s
Ran all test suites.
```


In the above failing snapshot, label is passed to `SelectComponent` but why isn't label expected?

File: src/Select.tsx [hunk 1]
```diff
@@ -92,6 +92,7 @@ export function Select(props: SelectProps) {
 						value={value}
 						onChange={onChange}
 						multiple={multiple}
+						label={!!label ? label : undefined}
```
Comment (lookfirst) [inline src/Select.tsx]: `!!label && undefined`

**Comment by lookfirst** [comment]
I'm not really sure how to fix the tests... I'll need your help with that. =)

File: src/Select.tsx (hunk 1 above)
Comment (afzalsayed96) [reply src/Select.tsx]: Uh.. `Select.test.tsx` passes label prop as `label={label && 'Test'}` which evaluates to `'Test'` for truthy and `false` when falsy. Hence, the explicit undefined. IMO this should be simplified as:

src/Select.tsx
```
label={label}
```

test/Select.test.tsx
```
label={label ? label : undefined}
```

**Comment by afzalsayed96** [comment]
Hey,

IMO the test snapshots need to be updated with `yarn test -u` and we are good.

Before

Screen Shot 2020-03-12 at 10 20 28 PM

After

Screen Shot 2020-03-12 at 10 18 46 PM

File: test/Select.test.tsx [hunk 2]
```diff
@@ -49,7 +49,7 @@ describe('Select', () => {
 					render={({ handleSubmit }) => (
 						<form onSubmit={handleSubmit} noValidate data-testid="form">
 							<Select
-								label={label && 'Test'}
+								label={label ? 'Test' : undefined}
```
Comment (lookfirst) [inline test/Select.test.tsx]: Can you try `!!label && 'Test'`?

**Comment by lookfirst** [comment]
Your screenshots both look the same to me? 🤷‍♂

**Comment by afzalsayed96** [comment]
> Your screenshots both look the same to me? 🤷‍♂

Yeah they do. I meant to show that there's is no visual difference before and after the change.

File: test/Select.test.tsx (hunk 2 above)
Comment (afzalsayed96) [reply test/Select.test.tsx]: That would still evaluate to `false` which we don't want

Comment (lookfirst) [reply test/Select.test.tsx]: Ah... I'm being dumb.

**Comment by lookfirst** [comment]
@afzalsayed96 cool. thanks for the contribution! I'll get a release out soon.

**Pull request merged**